internal/rbd/rbd_util.go

```diff
@@ -689,27 +692,41 @@ func genVolFromVolID(ctx context.Context, volumeID string, cr *util.Credentials,
 		return rbdVol, err
 	}
 
-	rbdVol.Pool, err = util.GetPoolName(rbdVol.Monitors, cr, vi.LocationID)
+	rbdVol.RadosNamespace, err = util.RadosNamespace(util.CsiConfigFile, rbdVol.ClusterID)
 	if err != nil {
-		return rbdVol, err
+		return nil, err
 	}
 
-	err = rbdVol.Connect(cr)
+	j, err := volJournal.Connect(rbdVol.Monitors, rbdVol.RadosNamespace, cr)
 	if err != nil {
 		return rbdVol, err
 	}
-	rbdVol.JournalPool = rbdVol.Pool
+	defer j.Destroy()
 
-	rbdVol.RadosNamespace, err = util.RadosNamespace(util.CsiConfigFile, rbdVol.ClusterID)
+	// check is there any volumeID mapping exists.
+	id, err := j.CheckNewUUIDMapping(ctx, rbdVol.JournalPool, volumeID)
 	if err != nil {
-		return nil, err
+		return rbdVol, fmt.Errorf("failed to get volume id %s mapping %s",
+			volumeID, err)
```

Comment:
use `%w` in the `Errorf()` call please. Mostly the format is like `"local message with local variables: original error"`